aviatrix/data_source_aviatrix_network_domains_test.go

```diff
@@ -0,0 +1,122 @@
+package aviatrix
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/AviatrixSystems/terraform-provider-aviatrix/v2/goaviatrix"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceAviatrixNetworkDomains_basic(t *testing.T) {
+	rName := acctest.RandString(5)
+	charset := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	tgwName := acctest.RandStringFromCharSet(5, charset) + acctest.RandString(5)
+	awsSideAsNumber := "64512"
+	ndName := acctest.RandStringFromCharSet(5, charset) + acctest.RandString(5)
+	resourceName := "data.aviatrix_network_domains.test"
+
+	skipAcc := os.Getenv("SKIP_DATA_NETWORK_DOMAINS")
+	if skipAcc == "yes" {
+		t.Skip("Skipping Data Source All Network Domains tests as SKIP_DATA_NETWORK_DOMAINS is set")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAviatrixNetworkDomainsConfigBasic(rName, tgwName, awsSideAsNumber, ndName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTgwNetworkDomainExists("aviatrix_aws_tgw_network_domain.test", tgwName, ndName),
+				),
+				Destroy: false,
+			},
+			{
+				Config: testAccDataSourceAviatrixNetworkDomainsConfigBasic(rName, tgwName, awsSideAsNumber, ndName),
```

Comment:
Is this second step required? You aren't doing anything different in the Config, can't you just move the Check part into the Step above?

Reply:
the second step is used for the get network_domains list, because this data source: 
data "aviatrix_network_domains" "test"{} will return null when create the network domain at the same time, so i think use the second step to refresh and read the data.

Reply:
use the "depends_on" to confirm the resource create after